File: server-scala-play-server/app/api/SubscriptionApi.scala
package api

import play.api.libs.json._
import model.ApplicationUsageResponse
import model.BigDecimal
import model.SirqulResponse
import model.SubscriptionPlanResponse
import model.SubscriptionResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait SubscriptionApi {
  /**
    * Create Subscription
    * Create a subscription for a billable entity.  Provide a planId, if not provided then the base plan will be assigned.
    * @param accountId The account used to perform the create, must be the responsible manager
    * @param planId The plan to subscribe to, if null use default plan
    * @param promoCode Set a promo code for a discount.
    */
  def createSubscription(version: BigDecimal, accountId: Long, planId: Option[Long], promoCode: Option[String]): SubscriptionResponse

  /**
    * Delete Subscription
    * Suspend the current subscription for the billable entity managed by the account.  The account must be the responsible manager to perform this action
    * @param accountId The account used to perform the delete, must be the responsible manager
    */
  def deleteSubscription(version: BigDecimal, accountId: Long): SirqulResponse

  /**
    * Get Subscription
    * Use the accountId to determine the associated BillableEntity.  Then get the subscription.
    * @param accountId The account used to perform the lookup
    */
  def getSubscription(version: BigDecimal, accountId: Long): SubscriptionResponse

  /**
    * Get Subscription Plan
    * Get the matched subscription plan
    * @param planId The ID of the plan to get
    */
  def getSubscriptionPlan(version: BigDecimal, planId: Long): SubscriptionPlanResponse

  /**
    * List Subscription Plans
    * Get the matched subscription plan
    * @param visible Include visible only (true), hidden only (false), or all (null)
    * @param role The role the plan is targeted for, values are: DEVELOPER, RETAILER, ADVERTISER
    */
  def getSubscriptionPlans(version: BigDecimal, visible: Option[Boolean], role: Option[String]): List[SubscriptionPlanResponse]

  /**
    * Get Subscription Usage
    * Use the accountId to determine the associated BillableEntity.  Then get the application usage.
    * @param accountId The account used to perform the lookup
    * @param applicationId Get for just 1 application instead of the BillableEntity
    * @param start The start time frame
    * @param end The end time frame
    */
  def getSubscriptionUsage(version: BigDecimal, accountId: Long, applicationId: Option[Long], start: Option[Long], end: Option[Long]): ApplicationUsageResponse

  /**
    * Update Subscription
    * Updates the subscription for the billable entity for an account
    * @param accountId The account used to perform the update, must be the responsible manager
    * @param planId The plan to subscribe to
    * @param promoCode Set a promo code for a discount.
    * @param active Set active status
    */
  def updateSubscription(version: BigDecimal, accountId: Long, planId: Option[Long], promoCode: Option[String], active: Option[Boolean]): SubscriptionResponse
}

File: server-ruby-sinatra/api/subscription_api.rb
require 'json'


MyApp.add_route('POST', '/api/3.18/subscription/create', {
  "resourcePath" => "/Subscription",
  "summary" => "Create Subscription",
  "nickname" => "create_subscription",
  "responseClass" => "SubscriptionResponse",
  "endpoint" => "/subscription/create",
  "notes" => "Create a subscription for a billable entity.  Provide a planId, if not provided then the base plan will be assigned.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "The account used to perform the create, must be the responsible manager",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "plan_id",
      "description" => "The plan to subscribe to, if null use default plan",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "promo_code",
      "description" => "Set a promo code for a discount.",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/3.18/subscription/delete', {
  "resourcePath" => "/Subscription",
  "summary" => "Delete Subscription",
  "nickname" => "delete_subscription",
  "responseClass" => "SirqulResponse",
  "endpoint" => "/subscription/delete",
  "notes" => "Suspend the current subscription for the billable entity managed by the account.  The account must be the responsible manager to perform this action",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "The account used to perform the delete, must be the responsible manager",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/subscription/get', {
  "resourcePath" => "/Subscription",
  "summary" => "Get Subscription",
  "nickname" => "get_subscription",
  "responseClass" => "SubscriptionResponse",
  "endpoint" => "/subscription/get",
  "notes" => "Use the accountId to determine the associated BillableEntity.  Then get the subscription.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "The account used to perform the lookup",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/subscription/plan/get', {
  "resourcePath" => "/Subscription",
  "summary" => "Get Subscription Plan",
  "nickname" => "get_subscription_plan",
  "responseClass" => "SubscriptionPlanResponse",
  "endpoint" => "/subscription/plan/get",
  "notes" => "Get the matched subscription plan",
  "parameters" => [
    {
      "name" => "plan_id",
      "description" => "The ID of the plan to get",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/subscription/plan/list', {
  "resourcePath" => "/Subscription",
  "summary" => "List Subscription Plans",
  "nickname" => "get_subscription_plans",
  "responseClass" => "Array<SubscriptionPlanResponse>",
  "endpoint" => "/subscription/plan/list",
  "notes" => "Get the matched subscription plan",
  "parameters" => [
    {
      "name" => "visible",
      "description" => "Include visible only (true), hidden only (false), or all (null)",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "role",
      "description" => "The role the plan is targeted for, values are: DEVELOPER, RETAILER, ADVERTISER",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/subscription/usage/get', {
  "resourcePath" => "/Subscription",
  "summary" => "Get Subscription Usage",
  "nickname" => "get_subscription_usage",
  "responseClass" => "ApplicationUsageResponse",
  "endpoint" => "/subscription/usage/get",
  "notes" => "Use the accountId to determine the associated BillableEntity.  Then get the application usage.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "The account used to perform the lookup",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "application_id",
      "description" => "Get for just 1 application instead of the BillableEntity",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "The start time frame",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "_end",
      "description" => "The end time frame",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/3.18/subscription/update', {
  "resourcePath" => "/Subscription",
  "summary" => "Update Subscription",
  "nickname" => "update_subscription",
  "responseClass" => "SubscriptionResponse",
  "endpoint" => "/subscription/update",
  "notes" => "Updates the subscription for the billable entity for an account",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "The account used to perform the update, must be the responsible manager",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "plan_id",
      "description" => "The plan to subscribe to",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "promo_code",
      "description" => "Set a promo code for a discount.",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "active",
      "description" => "Set active status",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

